thirdparty/datastore2/datastore_closer.go

```diff
@@ -33,3 +34,8 @@ func (w *datastoreCloserWrapper) Batch() (datastore.Batch, error) {
 
 	return bds.Batch()
 }
+
+// TODO(geoah) Move to ./testutils or somewhere else
+func ThreadSafeCloserMapDatastore() ThreadSafeDatastoreCloser {
+	return CloserWrap(syncds.MutexWrap(datastore.NewMapDatastore()))
+}
```

Comment:
Thats fine with me. The more we break things out the easier it will become to find homes for these sorts of things.